### ISSUE TITLE
Add support for reprozip

### ISSUE THREAD

**Issue by yuvipanda** [jupyter/repo2docker#133]
[ReproZip](https://www.reprozip.org/) is an awesome OSS project that lets users capture dependencies of a computation with syscall capture and turn that into a self contained `.rpz` file.  This `.rpz` file can then be built into a docker image & run!

We should add support for running .rpz files in repo2docker (and hence binderhub). There are two steps:

1. Figure out how to tell repo2docker that it's dealing with a rpz file (related to #102 refactoring maybe?)
2. Call out into [reprozip-docker](https://github.com/ViDA-NYU/reprozip/tree/1.0.x/reprounzip-docker) to produce a docker image

This should help us support rpz files in binderhub easily.

**Comment by choldgraf** [comment]
+1 to that, would be very cool. Maybe we can grab one of the reprozip people tomorrow and sit down w/ em? You were speaking to them after the talk, yeah?

**Comment by remram44** [comment]
We are around. In fact, you *were* sitting with us at breakfast 😝

**Comment by yuvipanda** [comment]
I had more conversations over lunch yesterday about this. Problems we need to solve:

1. Right now, repo2docker treats one repo (directory) as one image. With rpz files, one file is an image. How do we map this?
2. rpz files will most likely not have jupyter installed in them. How do we do that? We can't depend on any environment existing inside the rpz, so can't just depend on pip install working.
3. Same as (2) but for kernelspec files. How do we get those in? How do we decide what kernels need to be there?

This is just for the use case of 'rpz files opened with Jupyter Notebooks'. However, it is possible that the model instead should be just 'rpz files executing the one command / series of commands that were traced with them'. That's also a use-case we should support, and that will be conceptually a bit simpler perhaps. Am not sure what a non-API UI for that will look like yet, however.

**Comment by betatim** [comment]
Could we support "rpz file" as a new build pack? So you add the contents of one "on top of" the basic layers of the standard docker image used by repo2docker?

**Comment by choldgraf** [comment]
I think a challenge that Yuvi mentioned is that they often have repos with
multiple rpz files. But that doesn't seem like a deal breaker as you could
always use a text file to point to one or another.

**Comment by jzf2101** [comment]
@remram44 maybe we can talk about this some time?

**Comment by remram44** [comment]
Absolutely! We're happy to talk more about what you think we should look into and what you are interested in :wink:

I've been meaning to put forward a proof-of-concept of this, but I am pretty swamped. Hopefully very soon.

**Comment by yuvipanda** [comment]
What is a possible next step here? :)

**Comment by manics** [comment]
Is this still on the wish list?
